src/udev/net/link-config.c

```diff
@@ -380,11 +381,17 @@ int link_config_apply(link_config_ctx *ctx, link_config *config,
         if (!old_name)
                 return -EINVAL;
 
-        r = ethtool_set_speed(&ctx->ethtool_fd, old_name, config->speed / 1024, config->duplex);
-        if (r < 0)
-                log_warning_errno(r, "Could not set speed or duplex of %s to %zu Mbps (%s): %m",
-                                  old_name, config->speed / 1024,
-                                  duplex_to_string(config->duplex));
+        r = ethtool_set_glinksettings(&ctx->ethtool_fd, old_name, DIV_ROUND_UP(config->speed, 1000000), config->duplex, config->autoneg);
```

Comment:
I think you should define a variable for the rounded-up config speed. It seems silly to repeat the same formula three times.